storage/series/buffer.go

```diff
@@ -226,6 +232,33 @@ func bucketNeedsDrain(now time.Time, b *dbBuffer, idx int, start time.Time) int
 	return 0
 }
 
+func (b *dbBuffer) Tick() bufferTickResult {
+	// Avoid capturing any variables with callback
+	mergedOutOfOrder := b.computedForEachBucketAsc(computeAndResetBucketIdx, bucketTick)
+	return bufferTickResult{
+		mergedOutOfOrderBlocks: mergedOutOfOrder,
+	}
+}
+
+func bucketTick(now time.Time, b *dbBuffer, idx int, start time.Time) int {
+	// Perform a drain and reset if necessary
+	mergedOutOfOrderBlocks := bucketDrainAndReset(now, b, idx, start)
+
+	if b.buckets[idx].needsMerge() {
+		// Try to merge any out of order encoders to amortize the cost of a drain
+		r, err := b.buckets[idx].merge()
+		if err != nil {
+			log := b.opts.InstrumentOptions().Logger()
+			log.Errorf("buffer merge encode error: %v", err)
```

Comment:
would it be useful to log the related series here too?

Reply:
Because buffer is a composition of a series (i.e. series has a buffer) and there is no cyclical link back there is no way to access that ID unless we add an ID ptr to the buffer too.  This may be a little expensive considering that is yet another 16 bytes per series to add.  In all honesty it should never happen unless the encoder is unable to encode already encoded data in the streams.  There is a pretty serious application error and if we ever encountered this we'd need to find, fix the bug and completely end of life that version.

Reply:
Yeah fair. Only matters if finding which ID caused this would be particularly helpful in repro, sounds like we'd have bigger problems on hand than a single ID if this were to happen.